src/components/molecules/EventDisplay/EventDisplay.tsx

```diff
@@ -53,11 +48,26 @@ export const EventDisplay: React.FC<EventDisplayProps> = ({ event, venue }) => {
           {event.description}
         </div>
         <div className="schedule-event-info-room">
-          <div onClick={enterEvent}>
-            {event.room ?? "Enter"} {venue && `- ${venue.name}`}
-          </div>
+          {event.room && room ? (
+            <EnterRoomButton room={room as Room}>{buttonText}</EnterRoomButton>
```

Comment:
I don't feel that this is the best way to solve this. I think it is better than before, but can be improved. I'm opened to suggestions

Reply:
`as` assignment was used because we still have `Camp room` type or similar. Also, the way I separated this logic into a separate component doesn't seem to be perfect

Reply:
It might be worth adding an `@debt` comment or similar in these cases, that briefly explains what it is/why it was done that way/how/when it can be fixed up, eg.

```
// @debt using 'as' here because we still have multiple legacy Room types, clean up once we remove those
```